FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -0,0 +1,1054 @@
+#endif
+
+/* Check that the pre-signed GET URL is defined. */
+#ifndef democonfigS3_PRESIGNED_GET_URL
+    #error "Please define democonfigS3_PRESIGNED_GET_URL in demo_config.h."
+#endif
+
+/* Check that a TLS port for AWS IoT Core is defined. */
+#ifndef democonfigHTTPS_PORT
+    #define democonfigHTTPS_PORT    ( 443 )
+#endif
+
+/* Check the the queue size is defined. */
+#ifndef democonfigQUEUE_SIZE
+    #define democonfigQUEUE_SIZE    ( 20 )
+#endif
+
+/* Check that a transport timeout for transport send and receive is defined. */
+#ifndef democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS
+    #define democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS    ( 5000 )
+#endif
+
+/* Check that a size for the user buffer is defined. */
+#ifndef democonfigUSER_BUFFER_LENGTH
+    #define democonfigUSER_BUFFER_LENGTH    ( 4096 )
+#endif
+
+/* Check that the range request length is defined. */
+#ifndef democonfigRANGE_REQUEST_LENGTH
+    #define democonfigRANGE_REQUEST_LENGTH    ( 2048 )
+#endif
+
+/* Check that the stack size to use for HTTP tasks is defined. */
+#ifndef httpexampleTASK_STACK_SIZE
+    #define httpexampleTASK_STACK_SIZE    ( configMINIMAL_STACK_SIZE * 4 )
+#endif
+
+/**
+ * @brief Length of the pre-signed GET URL defined in demo_config.h.
+ */
+#define httpexampleS3_PRESIGNED_GET_URL_LENGTH               ( sizeof( democonfigS3_PRESIGNED_GET_URL ) - 1 )
+
+/**
+ * @brief The length of the HTTP GET method.
+ */
+#define httpexampleHTTP_METHOD_GET_LENGTH                    ( sizeof( HTTP_METHOD_GET ) - 1 )
+
+/**
+ * @brief Field name of the HTTP range header to read from the server response.
+ */
+#define httpexampleHTTP_CONTENT_RANGE_HEADER_FIELD           "Content-Range"
+
+/**
+ * @brief Length of the HTTP range header field.
+ */
+#define httpexampleHTTP_CONTENT_RANGE_HEADER_FIELD_LENGTH    ( sizeof( httpexampleHTTP_CONTENT_RANGE_HEADER_FIELD ) - 1 )
+
+/**
+ * @brief The HTTP status code returned for partial content.
+ */
+#define httpexampleHTTP_STATUS_CODE_PARTIAL_CONTENT          206
+
+/**
+ * @brief Ticks to wait for task notifications.
+ */
+#define httpexampleDEMO_TICKS_TO_WAIT                        pdMS_TO_TICKS( 1000 )
+
+/**
+ * @brief Notification bit indicating a prvReadFileSize() error in the response task.
+ */
+#define httpexampleFILE_SIZE_FAILURE                         ( 1U << 1 )
+
+/**
+ * @brief Notification bit indicating request task completion.
+ */
+#define httpexampleREQUEST_TASK_COMPLETION                   ( 1U << 2 )
+
+/**
+ * @brief Notification bit indicating response task completion.
+ */
+#define httpexampleRESPONSE_TASK_COMPLETION                  ( 1U << 3 )
+
+/**
+ * @brief The maximum number of loop iterations to wait after the last received
+ * server response, before declaring failure.
+ */
+#define httpexampleMAX_WAIT_ITERATIONS                       ( 10 )
+
+/**
+ * @brief Represents the network context used for the TLS session with the
+ * server.
+ */
+static NetworkContext_t xNetworkContext;
```

Comment:
All the S3 demos will have to be updated so that they build without errors:
https://github.com/FreeRTOS/FreeRTOS/commit/73b0d1b25930ac90c84c5cca49e50eeab8869a1d#diff-faf475d8aca710ebfc965f576f035deb85a4425193cdfbda0918b93a3f3c97ddR125-R227

Reply:
Thanks, will update.